mongorestore/restore.go

```diff
@@ -147,6 +147,7 @@ func (restore *MongoRestore) RestoreIntents() Result {
 			totalResult.combineWith(result)
 			if totalErr == nil && totalResult.Err != nil {
```

Comment:
After the change, this piece of logic is kinda convoluted. From my understanding, it seems you gonna return from the loop if there is any error and return with success if the function hits the bottom? 
So it can be simplified to
```
		for i := 0; i < restore.OutputOptions.NumParallelCollections; i++ {
			if totalErr == nil && totalResult.Err != nil {
			result := <-resultChan
                         totalResult.combineWith(result)
			if totalResult.Err != nil {
                               return totalResult
                         }
             }
             return totalResult
```
Please let me know if I understand correctly

Reply:
Yes, I believe we want to terminate the process after the first error. I'll update the file to reflect this simplification!